tests/plugins.test.ts

```diff
@@ -473,6 +483,120 @@ test('plugin config order', async () => {
+
+    await setupPlugins(mockServer)
+    const { pluginConfigs } = mockServer
+
+    const pluginConfig = pluginConfigs.get(39)!
+    expect(pluginConfig.plugin!.capabilities!.methods!.sort()).toEqual(['onEvent', 'processEvent', 'processEventBatch'])
+    expect(pluginConfig.plugin!.capabilities!.jobs).toEqual(['x'])
+    expect(pluginConfig.plugin!.capabilities!.scheduled_tasks).toEqual(['runEveryHour'])
+})
+
+test('plugin with source file loads capabilities', async () => {
+    const [plugin, unlink] = mockPluginTempFolder(`
+        function processEvent (event, meta) { event.properties={"x": 1}; return event }
+        function randomFunction (event, meta) { return event}
+        function onEvent (event, meta) { return event }
+    `)
+
+    getPluginRows.mockReturnValueOnce([plugin])
+    getPluginConfigRows.mockReturnValueOnce([pluginConfig39])
+    getPluginAttachmentRows.mockReturnValueOnce([pluginAttachment1])
+
+    await setupPlugins(mockServer)
+    const { pluginConfigs } = mockServer
+
+    const pluginConfig = pluginConfigs.get(39)!
+    expect(pluginConfig.plugin!.capabilities!.methods!.sort()).toEqual(['onEvent', 'processEvent', 'processEventBatch'])
+    expect(pluginConfig.plugin!.capabilities!.jobs).toEqual([])
+    expect(pluginConfig.plugin!.capabilities!.scheduled_tasks).toEqual([])
+
+    unlink()
+})
+
+test('plugin with source code loads capabilities', async () => {
+    const source_code = `
+        function processEvent (event, meta) { event.properties={"x": 1}; return event }
+        function randomFunction (event, meta) { return event}
+        function onSnapshot (event, meta) { return event }
+    `
+    getPluginRows.mockReturnValueOnce([mockPluginSourceCode(source_code)])
+
+    getPluginConfigRows.mockReturnValueOnce([pluginConfig39])
+    getPluginAttachmentRows.mockReturnValueOnce([pluginAttachment1])
+
+    await setupPlugins(mockServer)
+    const { pluginConfigs } = mockServer
+
+    const pluginConfig = pluginConfigs.get(39)!
+
+    expect(pluginConfig.plugin!.capabilities!.methods!.sort()).toEqual([
+        'onSnapshot',
+        'processEvent',
+        'processEventBatch',
+    ])
+    expect(pluginConfig.plugin!.capabilities!.jobs).toEqual([])
+    expect(pluginConfig.plugin!.capabilities!.scheduled_tasks).toEqual([])
+})
+
+test("capabilities don't reload without changes", async () => {
+    getPluginRows.mockReturnValue([plugin60])
+    getPluginAttachmentRows.mockReturnValue([pluginAttachment1])
+    getPluginConfigRows.mockReturnValue([pluginConfig39])
+
+    await setupPlugins(mockServer)
+    const pluginConfig = mockServer.pluginConfigs.get(39)!
+
+    pluginConfig.updated_at = new Date().toISOString()
+    // config is changed, but capabilities haven't changed
+
+    await setupPlugins(mockServer)
+    const newPluginConfig = mockServer.pluginConfigs.get(39)!
+
+    expect(newPluginConfig.plugin!.capabilities).toBe(pluginConfig.plugin!.capabilities)
```

Comment:
Let's also check that:

```ts
    expect(newPluginConfig.plugin).not.toBe(pluginConfig.plugin)
```

otherwise the capabilities are the same because it's literally the same object :)